Use the maven-publish plugin instead of the removed maven

After migrating to `Gradle 7`, the build started failing with the following exception
```
FAILURE: Build failed with an exception.

* Where:
Build file '/home/circleci/workspace/build.gradle' line: 81

* What went wrong:
A problem occurred evaluating root project '**************'.
> Plugin with id 'maven' not found
```
Looks like the `maven` plugin has been removed in Gradle 7. The official recommendation is to use use the `maven-publish` plugin instead

Source: https://docs.gradle.org/7.0/userguide/upgrading_version_6.html?_ga=2.108166854.555514338.1631315534-1494425579.1631315534#removal_of_the_legacy_maven_plugin

--- a/build.gradle
+++ b/build.gradle
@@ -78,7 +78,7 @@ subprojects {
   apply plugin: 'java'
   apply plugin: 'java-library'
   apply plugin: 'checkstyle'
-  apply plugin: 'maven'
+  apply plugin: 'maven-publish'
   apply plugin: "com.github.spotbugs"
   apply plugin: 'jacoco'
 

